clients/src/main/java/org/apache/kafka/clients/producer/MockProducer.java

```diff
@@ -146,8 +166,14 @@ private long nextOffset(TopicPartition tp) {
     }
 
     public synchronized void flush() {
-        while (!this.completions.isEmpty())
+        while (!this.completions.isEmpty()) {
             completeNext();
+        }
+        flushed = true;
+    }
+
+    public boolean flushed() {
```

Comment:
This function seems not used? Shouldn't we check if it is called in the unit tests?